In [1]:
#Importing the relevant libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [5]:
# Loading the data

Tts = pd.read_csv('data/Terry_Stops_20240826.csv')

# Checking the first few rows of the data
Tts.head()


Subject Age Group  Subject ID     GO / SC Num  Terry Stop ID  \
0           36 - 45          -1  20160000398323         208373   
1           18 - 25          -1  20180000227180         559146   
2           18 - 25          -1  20180000410091         498246   
3                 -          -1  20160000001637         146742   
4           46 - 55          -1  20150000006037         104477   

         Stop Resolution Weapon Type Officer ID  Officer YOB Officer Gender  \
0         Offense Report         NaN       4852         1953              M   
1  Citation / Infraction         NaN       5472         1964              M   
2         Offense Report         NaN       6081         1962              M   
3          Field Contact         NaN       6924         1974              M   
4          Field Contact         NaN       6732         1975              M   

  Officer Race  ...     Reported Time  \
0        Asian  ...  15:18:00.0000000   
1        Asian  ...  00:07:00.0000000   
2        White  ...  02:56:00.0000000   
3        White  ...  01:44:00.0000000   
4        White  ...  02:59:00.0000000   

                                  Initial Call Type  \
0                                          TRESPASS   
1           SUSPICIOUS PERSON, VEHICLE, OR INCIDENT   
2  NARCOTICS - VIOLATIONS (LOITER, USE, SELL, NARS)   
3                                                 -   
4                                                 -   

                            Final Call Type Call Type  \
0  --SUSPICIOUS CIRCUM. - SUSPICIOUS PERSON       911   
1             --TRAFFIC - BICYCLE VIOLATION       911   
2  --SUSPICIOUS CIRCUM. - SUSPICIOUS PERSON       911   
3                                         -         -   
4                                         -         -   

                              Officer Squad Arrest Flag Frisk Flag   Precinct  \
0     NORTH PCT 2ND W - LINCOLN - PLATOON 1           N          N      North   
1    SOUTHWEST PCT - 3RD WATCH - F/W RELIEF           N          N  Southwest   
2  NORTH PCT 3RD W - BOY (JOHN) - PLATOON 1           N          N      North   
3                  EAST PCT OPS - NIGHT ACT           N          N       East   
4         SOUTH PCT 3RD W - SAM - PLATOON 2           N          N      North   

  Sector Beat  
0      L   L3  
1      F   F3  
2      B   B1  
3      C   C1  
4      B   B2  

[5 rows x 23 columns]

In [7]:
Tts.shape

(61009, 23)

# Column Names and Descriptions
The following descriptions were provided by data.seattle.gov
This dataset contains the following data:

**Subject Age Group**: Subject Age Group (10 year increments) as reported by the officer.

**Subject ID**: Key, generated daily, identifying unique subjects in the dataset using a character to character match of first name and last name. "Null" values indicate an "anonymous" or "unidentified" subject. Subjects of a Terry Stop are not required to present identification.

**GO / SC Num**: General Offense or Street Check number, relating the Terry Stop to the parent report. This field may have a one to many relationship in the data.

**Terry Stop ID**: Key identifying unique Terry Stop reports.

**Stop Resolution**: Resolution of the stop as reported by the officer.

**Weapon Type**: Type of weapon, if any, identified during a search or frisk of the subject. Indicates "None" if no weapons was found.

**Officer ID**: Key identifying unique officers in the dataset.

**Officer YOB**: Year of birth, as reported by the officer.

**Officer Gender**: Gender of the officer, as reported by the officer.

**Officer Race**: Race of the officer, as reported by the officer.

**Subject Perceived Race**: Perceived race of the subject, as reported by the officer.

**Subject Perceived Gender**: Perceived gender of the subject, as reported by the officer.

**Reported Date**: Date the report was filed in the Records Management System (RMS). Not necessarily the date the stop occurred but generally within 1 day.

**Reported Time**: Time the stop was reported in the Records Management System (RMS). Not the time the stop occurred but generally within 10 hours.

**Initial Call Type**: Initial classification of the call as assigned by 911.

**Final Call Type**: Final classification of the call as assigned by the primary officer closing the event.

**Call Type**: How the call was received by the communication center.

**Officer Squad**: Functional squad assignment (not budget) of the officer as reported by the Data Analytics Platform (DAP).

**Arrest Flag**: Indicator of whether a "physical arrest" was made, of the subject, during the Terry Stop. Does not necessarily reflect a report of an arrest in the Records Management System (RMS).

**Frisk Flag**: Indicator of whether a "frisk" was conducted, by the officer, of the subject, during the Terry Stop.

**Precinct**: Precinct of the address associated with the underlying Computer Aided Dispatch (CAD) event. Not necessarily where the Terry Stop occurred.

**Sector**: Sector of the address associated with the underlying Computer Aided Dispatch (CAD) event. Not necessarily where the Terry Stop occurred.

**Beat**: Beat of the address associated with the underlying Computer Aided Dispatch (CAD) event. Not necessarily where the Terry Stop occurred.